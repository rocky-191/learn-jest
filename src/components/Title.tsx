import React,{CSSProperties,FC} from 'react';

interface Props {
  type:'large'| 'small';
  title:string;
}

const largeStyle:CSSProperties={
  fontSize:'2em',
  color:"red"
}

const smallStyle:CSSProperties={
  fontSize:'0.5em',
  color:'green'
}

const styleMapper:Record<'small'|'large',CSSProperties>={
  small:smallStyle,
  large:largeStyle
}

function Title(props:Props) {
  const {title,type}=props
  return (
    <>
      <p style={styleMapper[type]}>{title}</p>
    </>
  );
}

export default Title;