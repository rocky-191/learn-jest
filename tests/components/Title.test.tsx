import * as React from 'react';
import {render} from '@testing-library/react';
import Title from '@/components/Title';

describe('Title component',()=>{
  it('正确渲染大字',()=>{
    const {baseElement}=render(<Title type='large' title='大字' />);
    expect(baseElement).toMatchSnapshot();
  })

  it('正确渲染小字',()=>{
    const {baseElement}=render(<Title type='small' title='小字1' />);
    // 如果snapshot不通过，可使用 npx jest --updateSnapshop 命令更新快照
    expect(baseElement).toMatchSnapshot();
  })
})