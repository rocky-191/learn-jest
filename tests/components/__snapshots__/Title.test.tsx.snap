// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component 正确渲染大字 1`] = `
<body>
  <div>
    <p
      style="font-size: 2em; color: red;"
    >
      大字
    </p>
  </div>
</body>
`;

exports[`Title component 正确渲染小字 1`] = `
<body>
  <div>
    <p
      style="font-size: 0.5em; color: green;"
    >
      小字1
    </p>
  </div>
</body>
`;
